ui/app/components/app/permission-page-container/permission-page-container-content/permission-page-container-content.component.js

```diff
@@ -99,26 +102,93 @@ export default class PermissionPageContainerContent extends PureComponent {
+  }
+
+  renderAccountTooltip (textContent) {
+    const { selectedIdentities } = this.props
+    const { t } = this.context
+
+    return (
+      <Tooltip
+        key="all-account-connect-tooltip"
+        position="bottom"
+        wrapperClassName="permission-approval-container__bold-title-elements"
+        html={(
+          <div style={{ display: 'flex', flexDirection: 'column' }}>
+            { selectedIdentities.slice(0, 6).map((identity, index) => {
+              return (
+                <div key={ `tooltip-identity-${index}` }>
+                  { this.getAccountDescriptor(identity) }
+                </div>
+              )
+            }) }
+            { selectedIdentities.length > 6
+              ? t('plusXMore', [ selectedIdentities.length - 6 ])
+              : null
+            }
+          </div>
+        )}
+      >{ textContent }
+      </Tooltip>
+    )
+  }
+
+  renderTitleSubstituteText (name, key = 'title-substitute-text') {
+    return (
+      <div
+        key={key}
+        className="permission-approval-container__bold-title-elements"
+      >
+        { name }
+      </div>
+    )
+  }
+
+  getTitleArgs () {
+    const { domainMetadata, redirect, permissionRejected, selectedIdentities, allIdentitiesSelected } = this.props
     const { t } = this.context
 
-    let titleArgs
     if (redirect && permissionRejected) {
-      titleArgs = [ 'cancelledConnectionWithMetaMask' ]
+      return [ 'cancelledConnectionWithMetaMask' ]
     } else if (redirect) {
-      titleArgs = [ 'connectingWithMetaMask' ]
+      return [ 'connectingWithMetaMask' ]
     } else if (domainMetadata.extensionId) {
-      titleArgs = [ 'externalExtension', [domainMetadata.extensionId] ]
+      return [ 'externalExtension', [domainMetadata.extensionId] ]
+    } else if (allIdentitiesSelected) {
+      return [
+        'connectToAll',
+        [
+          <div
+            key="multi-account-connect-all-accounts"
+            className="permission-approval-container__bold-title-elements"
+          >
+            { this.renderAccountTooltip(t('connectToAllAccounts')) }
+          </div>,
+        ],
+      ]
     } else if (selectedIdentities.length > 1) {
-      titleArgs = ['connectToMultiple', [ selectedIdentities.length ] ]
+      return [
+        'connectToMultiple',
+        [
+          this.renderAccountTooltip(t('connectToMultipleNumberOfAccounts', [ selectedIdentities.length ])),
+        ],
+      ]
     } else {
-      titleArgs = [
+      return [
         'connectTo', [
-          `${selectedIdentities[0].label} (...${selectedIdentities[0].address.slice(selectedIdentities[0].address.length - 4)})`,
+          this.renderTitleSubstituteText(this.getAccountDescriptor(selectedIdentities[0]), 'title-substitute-text-account-name'),
         ],
       ]
     }
+  }
+
+  render () {
+    const { domainMetadata, redirect } = this.props
+    const { t } = this.context
+
+    const titleArgs = this.getTitleArgs()
```

Comment:
Nit: This is not important, and it may be out of scope of this PR (it was done in a previous PR), but as a general rule it would be nice to avoid building up sets of arguments to pass to the `t` function. Instead we could call `t` directly here, and get the title rather than the title args.

This would be preferred because it'd make it possible to more reliably lint the codebase to ensure there are no unused localized messages. Currently that's not possible. This is on my "maybe someday" todo list.